src/main/scala/tech/cryptonomic/conseil/tezos/TezosRemotes.scala

```diff
@@ -0,0 +1,202 @@
+package tech.cryptonomic.conseil.tezos
+
+import tech.cryptonomic.conseil.config.{HttpStreamingConfiguration, NetworkCallsConfiguration}
+import tech.cryptonomic.conseil.config.Platforms.TezosConfiguration
+import tech.cryptonomic.conseil.generic.chain.RemoteRpc
+import tech.cryptonomic.conseil.util.JsonUtil.JsonString
+
+
+object TezosRemoteInstances {
+
+  object Akka {
+    import com.typesafe.scalalogging.Logger
+    import cats.Monoid
+    import scala.concurrent.Future
+    import akka.actor.ActorSystem
+    import akka.stream.ActorMaterializer
+    import akka.stream.scaladsl.Source
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model._
+
+    case class RemoteContext(
+      tezosConfig: TezosConfiguration,
+      requestConfig: NetworkCallsConfiguration,
+      streamingConfig: HttpStreamingConfiguration
+    )(implicit val system: ActorSystem) {
+
+      object ShutdownComplete extends ShutdownComplete
+      private val rejectingCalls = new java.util.concurrent.atomic.AtomicBoolean(false)
+
+      def translateCommandToUrl(command: String): String = {
+        import tezosConfig.nodeConfig._
+        s"$protocol://$hostname:$port/${pathPrefix}chains/main/$command"
+      }
+
+      def shutdown(): Future[ShutdownComplete] = {
+        rejectingCalls.compareAndSet(false, true)
+        Http(system).shutdownAllConnectionPools().map(_ => this.ShutdownComplete)(system.dispatcher)
+      }
+
+      def withRejectionControl[T, Container[_]](call: => Container[T])(implicit mono: Monoid[Container[T]]): Container[T] =
+        if (rejectingCalls.get) mono.empty else call
+
+    }
+
+    trait Futures {
+      import cats.Id
+      import cats.data.Const
+      import cats.instances.future._
+      import cats.syntax.apply._
+      import scala.concurrent.ExecutionContext
+      import scala.util.control.NoStackTrace
+
+      //might be actually unlawful, but we won't use it for append, only for empty
+      private implicit def futureMonoid[T](implicit ec: ExecutionContext) = new Monoid[Future[T]] {
+        override def combine(x: Future[T], y: Future[T]) = x *> y
+        override def empty: Future[T] = Future.failed(new IllegalStateException("Tezos node requests will no longer be accepted.") with NoStackTrace)
+      }
+
+      type JustString[CallId] = Const[String, CallId]
+
+      implicit def futuresInstance(implicit context: RemoteContext) =
+        new RemoteRpc[Future, Id, JustString] {
+          import context._
+
+          private val logger = Logger("Akka.Futures.RemoteRpc")
+
+          implicit val system = context.system
+          implicit val dispatcher = system.dispatcher
+          implicit val materializer = ActorMaterializer()
+
+          type CallConfig = Any
+          type PostPayload = JsonString
+
+          def runGetCall[CallId](
+            callConfig: Any = (),
```

Comment:
Why not just a `Unit`?

Reply:
because here we're actually implementing the `trait`'s interface.

It should more aptly be defined as `override def runGetCall ...`

The method type signature is already set in stone. Since this parameter is actually ignored by this instance, the choice is to use a completely general input type (i.e. Any), so that you can pass whatever you want to the method call, as it will be ignored anyway.
As a default argument value, we could've used literally anything, so I picked `()` as the more memory efficient object to pass in and ignore.

To sum it up, if I changed `callConfig` to expect `Unit` the signature would not have matched and the `futureInstance` class would've failed to implement `RemoteRpc`